test/integration/backward_compatible/parallel/serialization/compact/CompactPublicAPIsTest.js

```diff
@@ -0,0 +1,874 @@
+                result.should.be.deep.eq(OUTER_INSTANCE);
+            }
+        });
+
+        it('containsKey', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                await obj.containsKey(car1);
+            }
+        });
+
+        it('containsValue', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                await obj.containsValue(car1);
+            }
+        });
+
+        it('put', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                await obj.put(car1, employee);
+            }
+        });
+
+        it('putAll', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                await obj.putAll([[car1, car1], [employee, employee]]);
+            }
+        });
+
+        it('setAll', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                await obj.setAll([[car1, car1], [employee, employee]]);
+            }
+        });
+
+        it('get', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                await obj.get(car1);
+            }
+        });
+
+        it('getAll', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                await obj.getAll([car1, employee]);
+            }
+        });
+
+        it('remove', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                await obj.remove(car1, employee);
+            }
+        });
+
+        it('delete', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                await obj.delete(car1);
+            }
+        });
+
+        it('evict', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                await obj.evict(car1);
+            }
+        });
+
+        it('forceUnlock', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                await obj.forceUnlock(car1);
+            }
+        });
+
+        it('isLocked', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                await obj.isLocked(car1);
+            }
+        });
+
+        it('loadAll', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                const error = await TestUtil.getRejectionReasonOrThrow(
+                    async () => await obj.loadAll([car1, employee])
+                );
+                // MapStore configuration is needed for this to work, it throws NullPointerError.
+                // So we assert it does not throw SchemaNotReplicatedError
+                error.should.not.be.instanceOf(SchemaNotReplicatedError);
```

Comment:
minor: I believe asserting that this is NullPointerError is better

Reply:
You know my idea about this from python pr reviews. Let's do it like this for node.js and figure out which one is better in the future. I don't think my assertion is not enough because I am only testing serialization here.